cfme/physical/physical_server.py

```diff
@@ -0,0 +1,232 @@
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins.
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def execute_button(self, button_group, button, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button_group).item_select(button, handle_alert=handle_alert)
+
+    def power_on(self):
+        view = navigate_to(self, "Details")
+        view.toolbar.power.item_select("Power On", handle_alert=True)
+
+    def power_off(self):
+        view = navigate_to(self, "Details")
+        view.toolbar.power.item_select("Power Off", handle_alert=True)
+
+    @property
+    def power_state(self):
+        return self.get_detail("Properties", "Power State")
+
+    def refresh(self, cancel=False):
+        """Perform 'Refresh Relationships and Power States' for the server.
+
+        Args:
+            cancel (bool): Whether the action should be cancelled, default to False
+        """
+        view = navigate_to(self, "Details")
+        view.toolbar.configuration.item_select("Refresh Relationships and Power States",
+            handle_alert=cancel)
+
+    def wait_for_physical_server_state_change(self, desired_state, timeout=300):
+        """Wait for PhysicalServer to come to desired state. This function waits just the needed amount of
+           time thanks to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+        """
+        view = navigate_to(self.parent, "All")
+
+        def _looking_for_state_change():
+            entity = view.entities.get_entity(name=self.name)
+            return "currentstate-{}".format(desired_state) in entity.data['state']
+
+        wait_for(_looking_for_state_change, fail_func=view.browser.refresh, num_sec=timeout)
+
+    def get_detail(self, title, field):
+        """Gets details from the details summary tables.
+
+        Args:
+            title (str): Summary Table title
+            field (str): Summary table field name
+
+        Returns: A string representing the entities of the SummaryTable's value.
+        """
+        view = navigate_to(self, "Details")
+        return getattr(view.entities, title.lower().replace(" ", "_")).get_text_of(field)
+
+    @property
+    def exists(self):
+        """Checks if the physical_server exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    def get_db_id(self):
+        if self.db_id is None:
```

Comment:
This definitely looks like it would benefit from use of cached_property.

Reply:
@mshriver Could you give an example of where cached_property is used? Thanks

Reply:
Absolutely:
It's a pip project with pretty good simple examples:
https://pypi.python.org/pypi/cached-property

We use it throughout the codebase, with varying degrees of complexity:
https://github.com/ManageIQ/integration_tests/blob/067ec3d197a2afa2bca8ff4a06d4aeec2d9c415d/cfme/infrastructure/config_management.py#L540

The idea being its a property that doesn't really change, but can be invalidated when need be.